cli/src/lib.rs

```diff
@@ -512,31 +512,73 @@ fn override_toolchain(cfg_override: &ConfigOverride) -> Result<RestoreToolchainC
 
             Ok(version)
         }
+        
+        fn is_solana_version_installed(version: &str) -> Result<bool> {
+            let output = std::process::Command::new("ls")
+                .arg("-l")
+                .arg("~/.local/share/solana/install/releases")
+                .output()?;
+        
+            if output.status.success() {
+                let output_versions = std::str::from_utf8(&output.stdout)?;
+                Ok(output_versions.lines().any(|line| line.contains(version)))
```

Comment:
Using `line.contains` here can return false positive, e.g. specifying:

```toml
[toolchain]
solana_version = "1.17.1"
```

and having `1.17.14` installed would return `true` even if the `1.17.1` version is not installed.

That's why we parse the exact version in `get_current_version` function:

https://github.com/coral-xyz/anchor/blob/1a6b5066dfc52e4b60ba32d92059cafc4f5f637c/cli/src/lib.rs#L503-L511

Reply:
> Using `line.contains` here can return false positive, e.g. specifying:
> 
> ```toml
> [toolchain]
> solana_version = "1.17.1"
> ```
> 
> and having `1.17.14` installed would return `true` even if the `1.17.1` version is not installed.
> 
> That's why we parse the exact version in `get_current_version` function:
> 
> https://github.com/coral-xyz/anchor/blob/1a6b5066dfc52e4b60ba32d92059cafc4f5f637c/cli/src/lib.rs#L503-L511

Updated the function to be this : 
```rs
fn is_solana_version_installed(version: &str) -> Result<bool> {
            let output = std::process::Command::new("ls")
                .arg("-l")
                .arg("~/.local/share/solana/install/releases")
                .output()?;

            if output.status.success() {
                let output_versions = std::str::from_utf8(&output.stdout)?;

                let installed_versions: Vec<String> = output_versions
                    .lines()
                    .filter_map(|line| {
                        Regex::new(r"(\d+\.\d+\.\S+)")
                            .ok()
                            .and_then(|re| re.captures(line))
                            .map(|captures| captures[0].to_string())
                    })
                    .collect();

                Ok(installed_versions.contains(&version.to_string()))
            } else {
                Err(anyhow!("Failed to list Solana installed versions"))
            }
        }
```

this would do the work I'm pretty sure